test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/ViewComponentMethodSelectorTest.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    public class ViewComponentMethodSelectorTest
+    {
+        [Theory]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[0])]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[] { "" })]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[] { 42 })]
+        [InlineData(typeof(ViewComponentWithAsyncInvoke), new object[] { 42, false })]
```

Comment:
Yeah it started off being different. I can trim a case or two since they test the same behavior